### PR TITLE
Make RDBMS query timeout configurable

### PR DESCRIPTION
**Pull request by NelsonVides**
A strict hardcoded macro is too inflexible and this is a parameter I want to iterate on the configuration easily.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/131750) / small_tests / 77bf0cbf33c9deead7e31849f9468bafbd65569b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3728/131750/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3728/131750/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-10_12.41.59/lib.mongooseim.logs/run.2022-08-10_12.41.59/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/131779) / small_tests / a6902514b41c70ed8a66cbe9313cdf954469227e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3728/131779/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3728/131779/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-10_12.54.50/lib.mongooseim.logs/run.2022-08-10_12.54.50/suite.log.html)

File: src/config/mongoose_config_spec.erl [hunk 1]
```diff
@@ -555,6 +555,8 @@ outgoing_pool_connection(<<"rdbms">>) ->
                                          validate = {enum, [odbc, pgsql, mysql]}},
                  <<"keepalive_interval">> => #option{type = integer,
                                                      validate = positive},
+                 <<"query_timeout">> => #option{type = integer,
+                                                     validate = positive},
```
Comment (chrzaszcz) [inline src/config/mongoose_config_spec.erl:559]:
```suggestion
                                                validate = positive},
```

File: test/common/config_parser_helper.erl [hunk 2]
```diff
@@ -836,7 +836,7 @@ default_pool_conn_opts(redis) ->
       database => 0,
       password => ""};
 default_pool_conn_opts(rdbms) ->
-    #{max_start_interval => 30};
+    #{query_timeout => 5, max_start_interval => 30};
```
Comment (chrzaszcz) [inline test/common/config_parser_helper.erl:839]: Why 5 if the default is 5000?

**Review by chrzaszcz — COMMENTED** [review]
Please add a test case to `config_parser_SUITE`.

File: test/common/config_parser_helper.erl (hunk 2 above)
Comment (NelsonVides) [reply test/common/config_parser_helper.erl:839]: Confusing seconds with milliseconds 🤦🏽

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/131828) / small_tests / 57d3ba68befcfd6775b17bf5e21ee11c1389627e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3728/131828/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3728/131828/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-10_13.13.15/lib.mongooseim.logs/run.2022-08-10_13.13.15/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**